crates/sui-core/src/authority.rs

```diff
@@ -807,14 +807,26 @@ impl AuthorityState {
 
         // If commit_certificate returns an error, tx_guard will be dropped and the certificate
         // will be persisted in the log for later recovery.
-        self.commit_certificate(
-            inner_temporary_store,
-            certificate,
-            &signed_effects,
-            bypass_validator_halt,
-        )
-        .await
-        .tap_err(|e| error!(?digest, "commit_certificate failed: {}", e))?;
+        if let Err(err) = self
+            .commit_certificate(
+                inner_temporary_store,
+                certificate,
+                &signed_effects,
+                bypass_validator_halt,
+            )
+            .await
+        {
+            if matches!(err, SuiError::ValidatorHaltedAtEpochEnd) {
+                debug!(
```

Comment:
Just curious, how many TXes are expected to fall into this condition, of not being able to proceed because of a halt?  And is it necessary to halt right away, instead of letting them finish?

Reply:
We could wait a bit to reduce the chance of this happening, but at some point we need to close the door